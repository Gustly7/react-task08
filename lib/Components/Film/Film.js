'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _index = require('./Poster/index');

var _index2 = _interopRequireDefault(_index);

var _index3 = require('./FilmName/index');

var _index4 = _interopRequireDefault(_index3);

var _index5 = require('./FilmYear/index');

var _index6 = _interopRequireDefault(_index5);

var _index7 = require('./FilmGenre/index');

var _index8 = _interopRequireDefault(_index7);

require('./Film.css');

var _reactRouterDom = require('react-router-dom');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function itemClickHandler() {
  window.scrollTo(0, 0);
}

//class Film extends Component {

var Film = function Film(props) {

  render();
  {
    if (window['sessionStorage'] === undefined) {
      window['sessionStorage'] = {};
    }

    if (window['sessionStorage']['searchText'] === undefined) {
      window['sessionStorage']['searchText'] = ' ';
    }

    if (window['sessionStorage']['searchText'] === '') {
      throw new Error('Пустой поиск!');
    }
    console.log(undefined.props.film);
    return _react2.default.createElement(
      _reactRouterDom.Link,
      { to: '/film/' + undefined.props.film.id, onClick: function onClick() {
          return undefined.itemClickHandler();
        }, className: 'film-link' },
      _react2.default.createElement(
        'div',
        { className: 'Film' },
        _react2.default.createElement(
          'div',
          { className: 'container' },
          _react2.default.createElement(
            'div',
            { className: 'row' },
            _react2.default.createElement(
              'div',
              { className: 'col' },
              _react2.default.createElement(_index2.default, { link: undefined.props.film.poster_path })
            )
          ),
          _react2.default.createElement(
            'div',
            { className: 'row' },
            _react2.default.createElement(
              'div',
              { className: 'col' },
              _react2.default.createElement(_index4.default, { filmName: undefined.props.film.title })
            ),
            _react2.default.createElement(
              'div',
              { className: 'col' },
              _react2.default.createElement(_index6.default, { filmYear: undefined.props.film.release_date })
            )
          ),
          _react2.default.createElement(
            'div',
            { className: 'row' },
            _react2.default.createElement(
              'div',
              { className: 'col' },
              _react2.default.createElement(_index8.default, { filmGenre: undefined.props.film.genres.join(', ') })
            )
          )
        )
      )
    );
  }
};

exports.default = Film;