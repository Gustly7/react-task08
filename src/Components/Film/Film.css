.Poster img{
    width: 250px;
    height: 356.5px;
}
.FilmName{
    text-align: left;
    font-weight: bold;
}
.FilmYear{
    border: 1px;
    border-style: solid;
    text-align: center;
}
.FilmGenre{
    text-align: left;
    color: gray;
}
.Film{
    padding: 20px;
}