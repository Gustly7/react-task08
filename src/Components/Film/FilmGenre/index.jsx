import React, { Component } from 'react'

class FilmGenre extends Component {
  render () {
    return (
      <div className='FilmGenre'>
        {this.props.filmGenre}
      </div>
    )
  }
}

export default FilmGenre
