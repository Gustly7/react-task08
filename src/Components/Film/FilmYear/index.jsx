import React, { Component } from 'react'

class FilmYear extends Component {
  render () {
    return (
      <div className='FilmYear'>
        {this.props.filmYear.substr(0, 4)}
      </div>
    )
  }
}

export default FilmYear
