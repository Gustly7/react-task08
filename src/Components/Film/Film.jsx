// @flow
import React, { Component } from 'react'
import Poster from './Poster/index'
import FilmName from './FilmName/index'
import FilmYear from './FilmYear/index'
import FilmGenre from './FilmGenre/index'
import './Film.css'
import { Link } from 'react-router-dom'

type film = {
  id: number;
  poster_path: string;
  title: string;
  release_date: string;
  genres: Array<string>;
}

type Props = {
  className?: string,
  disabled?: boolean,
  film: film
};

class Film extends Component<Props> {
//const Film = (props: film) => {

  itemClickHandler (){
    window.scrollTo(0, 0)
  }

  render ()
  {
    if (window['sessionStorage'] === undefined) { window['sessionStorage'] = {} }

    if (window['sessionStorage']['searchText'] === undefined) { window['sessionStorage']['searchText'] = ' ' }

    if (window['sessionStorage']['searchText'] === '') {
      throw new Error('Пустой поиск!')
    }
    console.log(this.props.film)
    return (
      <Link to={'/film/' + this.props.film.id} onClick={() => this.itemClickHandler()} className='film-link'>
        <div className='Film'>
          <div className='container'>
            <div className='row'>
              <div className='col'>
                <Poster link={this.props.film.poster_path} />
              </div>
            </div>
            <div className='row'>
              <div className='col'>
                <FilmName filmName={this.props.film.title} />
              </div>
              <div className='col'>
                <FilmYear filmYear={this.props.film.release_date} />
              </div>
            </div>
            <div className='row'>
              <div className='col'>
                <FilmGenre filmGenre={this.props.film.genres.join(', ')} />
              </div>
            </div>
          </div>
        </div>
      </Link>
    )
  }
}

export default Film
