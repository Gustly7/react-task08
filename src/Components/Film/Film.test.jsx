import React, { Component } from 'react'
import Film from './Film.jsx'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

describe('Film component test suite', () => {
  window['sessionStorage'] = {}
  window['sessionStorage']['searchText'] = 'just film'

  test('Film snapshot renders correctly', () => {
    const tree = renderer.create(
      <Film films={{poster_path: '', release_date: '2000'}} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Film div should have poster, year, genre and name', () => {
    const wrapper = mount(
      <Film films={{poster_path: '', release_date: '2000'}} />
    )
    const elem = wrapper.find('div.Poster')
    expect(elem.length).toBe(1)
  })
})
