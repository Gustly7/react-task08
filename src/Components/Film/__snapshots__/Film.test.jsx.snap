// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film component test suite Film snapshot renders correctly 1`] = `
<div
  className="Film"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="Poster"
        >
          <img
            alt={undefined}
            src=""
          />
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="FilmName"
        />
      </div>
      <div
        className="col"
      >
        <div
          className="FilmYear"
        >
          2000
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="FilmGenre"
        />
      </div>
    </div>
  </div>
</div>
`;
